### PR TITLE
[#2150] Improvement : Arguments in incorrect order fixed with correct order

### PR DESCRIPTION
**Pull request by ajayl83**
What changes were proposed in this pull request?

- Arguments in incorrect order fixed with correct order!

Why are the changes needed?

- Fix: #2150 

Does this PR introduce any user-facing change?

- No

How was this patch tested?

- ./gradlew test -PskipITs

### CODE REVIEW

**Comment by justinmclean** [comment]
Hi it looks like you have more than one issue here can you revert the changes related to Distribution?

**Comment by ajayl83** [comment]
> Hi it looks like you have more than one issue here can you revert the changes related to Distribution?

oh yes , apologies i have reverted the commit

**Comment by justinmclean** [comment]
Thanks for fixing that.

**Comment by ajayl83** [comment]
@jerryshao i dont see existing test so can i add only for this method or all :)

**Comment by mchades** [comment]
It's better to add all tests for partitionDTO(`IdentityPartitionDTO`, `RangePartitionDTO` and `ListPartitionDTO`) in `com.datastrato.gravitino.json.TestJsonUtils`

**Comment by ajayl83** [comment]
@jerryshao @mchades i added the necessary tests however while running the build against jdk17 i am coming across below error, can you kindly let me know how to fix this?
 Task :rat FAILED
Files with unapproved licenses:
 - /Users/ajaylele/Downloads/gravitino/common/src/test/java/com/datastrato/gravitino/dto/util/DTOConvertersTest.java

**Comment by mchades** [comment]
You can push the codes, so that we can see detailed information from GitHub's CI.

**Comment by jerryshao** [comment]
> @jerryshao @mchades i added the necessary tests however while running the build against jdk17 i am coming across below error, can you kindly let me know how to fix this? Task :rat FAILED Files with unapproved licenses:
> 
> * /Users/ajaylele/Downloads/gravitino/common/src/test/java/com/datastrato/gravitino/dto/util/DTOConvertersTest.java

Seems like the license header issue, did you add the license header to this `DTOConvertersTest.java` file?

**Comment by ajayl83** [comment]
@jerryshao there is already license header, the other problem was i was using static import, i fixed it but still build fails with same problem :(

**Comment by ajayl83** [comment]
@jerryshao my bad i earlier wrote new UT for this, but have moved now to TestJsonUtils.java :) , its good now, will commit it

**Comment by jerryshao** [comment]
@mchades would you please help to review.

File: common/src/test/java/com/datastrato/gravitino/json/TestJsonUtils.java [hunk 1]
```diff
@@ -432,4 +439,89 @@ void testIndexDTOSerde() throws JsonProcessingException {
         objectMapper.readValue(expected, IndexDTO.class),
         objectMapper.readValue(jsonValue, IndexDTO.class));
   }
+
+  @Test
+  void testFromDTOPartitionDTOWhenIdentityPartitionDTOToPartition() {
```
Comment (mchades) [inline common/src/test/java/com/datastrato/gravitino/json/TestJsonUtils.java:444]: The function name can be simplified to `testIdentityPartitionDTOConvert`

File: common/src/test/java/com/datastrato/gravitino/json/TestJsonUtils.java
Comment (mchades) [inline common/src/test/java/com/datastrato/gravitino/json/TestJsonUtils.java:1]: Sorry for my misguided advice before, the new methods should be added in the new test class `com.datastrato.gravitino.dto.util.TestDTOConverters`

File: common/src/test/java/com/datastrato/gravitino/json/TestJsonUtils.java [hunk 2]
```diff
@@ -432,4 +439,89 @@ void testIndexDTOSerde() throws JsonProcessingException {
...
+    PartitionDTO identityPartitionDTO =
+        IdentityPartitionDTO.builder()
+            .withFieldNames(fieldNames)
+            .withName("IdentityPartition")
+            .withValues(values)
+            .withProperties(properties)
+            .build();
+    // when
+    IdentityPartition identityPartition =
+        (IdentityPartition) DTOConverters.fromDTO(identityPartitionDTO);
```
Comment (mchades) [inline common/src/test/java/com/datastrato/gravitino/json/TestJsonUtils.java:466]: both `fromDTO` and `toDTO` need to be tested

Comment (ajayl83) [reply common/src/test/java/com/datastrato/gravitino/json/TestJsonUtils.java]: @mchades added tests for toDTO and moved tests to new class

**Comment by ajayl83** [comment]
@mchades i am having weird issue with spotlessJava formatting , if i execute this all the imports are gone, can you kindly assist?

**Comment by ajayl83** [comment]
@mchades its my IDE problem, do we have a settings file that can be imported for Intellij?

**Comment by mchades** [comment]
FYI:
settings -> Editor -> Code Style -> Java -> Imports

then modify:
`Class count to use import with '*'` to 99
`Names count to use static import with '*` to 99

File: common/src/test/java/com/datastrato/gravitino/dto/util/TestDTOConverters.java [hunk 3]
```diff
@@ -0,0 +1,184 @@
+/*
+ * Copyright 2023 Datastrato Pvt Ltd.
```
Comment (mchades) [inline common/src/test/java/com/datastrato/gravitino/dto/util/TestDTOConverters.java:2]: 2023 -> 2024

Comment (ajayl83) [reply common/src/test/java/com/datastrato/gravitino/dto/util/TestDTOConverters.java]: @mchades done and thanks for settings!

**Review by mchades — APPROVED** [review]
LGTM

**Pull request merged**